src/SimpleRADOSStriper.h

```diff
@@ -0,0 +1,137 @@
+
+#include <string_view>
+#include <thread>
+
+#include "include/buffer.h"
+#include "include/rados/librados.hpp"
+#include "include/uuid.h"
+#include "include/types.h"
+
+#include "common/ceph_time.h"
+#include "common/perf_counters.h"
+
+class SimpleRADOSStriper
+{
+public:
+  using aiocompletionptr = std::unique_ptr<librados::AioCompletion>;
+  using clock = ceph::coarse_mono_clock;
+  using time = ceph::coarse_mono_time;
+
+  struct extent {
+    std::string soid;
+    size_t len;
+    size_t off;
+  };
+
+  static inline const uint64_t object_size = 22; /* power of 2 */
+  static inline const uint64_t min_growth = (1<<27); /* 128 MB */
+  static inline const char XATTR_EXCL[] = "striper.excl";
+  static inline const char XATTR_SIZE[] = "striper.size";
+  static inline const char XATTR_ALLOCATED[] = "striper.allocated";
+  static inline const char XATTR_VERSION[] = "striper.version";
+  static inline const char XATTR_LAYOUT_STRIPE_UNIT[] = "striper.layout.stripe_unit";
+  static inline const char XATTR_LAYOUT_STRIPE_COUNT[] = "striper.layout.stripe_count";
+  static inline const char XATTR_LAYOUT_OBJECT_SIZE[] = "striper.layout.object_size";
+  static inline const std::string biglock = "striper.lock";
+  static inline const std::string lockdesc = "SimpleRADOSStriper";
+
+  static int config_logger(CephContext* cct, std::string_view name, std::shared_ptr<PerfCounters>* l);
+
+  SimpleRADOSStriper() = default;
+  SimpleRADOSStriper(librados::IoCtx _ioctx, std::string _oid)
+    : ioctx(std::move(_ioctx))
+    , oid(std::move(_oid))
+  {
+    cookie.generate_random();
+    auto r = librados::Rados(ioctx);
+    myaddrs = r.get_addrs();
+  }
+  SimpleRADOSStriper(const SimpleRADOSStriper&) = delete;
+  SimpleRADOSStriper& operator=(const SimpleRADOSStriper&) = delete;
+  SimpleRADOSStriper& operator=(SimpleRADOSStriper&&) = delete;
+  SimpleRADOSStriper(SimpleRADOSStriper&&) = delete;
+  ~SimpleRADOSStriper();
+
+  int create();
+  int open();
+  int remove();
+  int stat(uint64_t* size);
+  ssize_t write(const void* data, size_t len, uint64_t off);
+  ssize_t read(void* data, size_t len, uint64_t off);
+  int truncate(size_t size);
+  int flush();
+  int lock(uint64_t timeoutms);
+  int unlock();
+  int is_locked() const {
+    return locked;
+  }
+  int printlockers(std::ostream& out);
+  void set_logger(std::shared_ptr<PerfCounters> l) {
+    logger = std::move(l);
+  }
+  void set_lock_interval(std::chrono::milliseconds t) {
+    lock_keeper_interval = t;
+  }
+  void set_lock_timeout(std::chrono::milliseconds t) {
+    lock_keeper_timeout = t;
+  }
+  void set_blocklist_the_dead(bool b) {
+    blocklist_the_dead = b;
+  }
+
+protected:
+  ceph::bufferlist str2bl(std::string_view sv);
+  ceph::bufferlist uint2bl(uint64_t v);
+  int setmeta(uint64_t new_size, bool update_size);
+  int allocshrink(uint64_t a);
+  int maybeallocshrink();
+  int wait_for_updates();
+  int recover_lock();
+  extent getnextextent(size_t len, uint64_t off) const;
```

Comment:
nit, i'd suggest add underscores between the words, and place `off` before `len`, like
```c++
get_next_extent(uint64_t off, size_t len)
```